### PR TITLE
Unknown theme elements throw warnings instead of errors

### LINKED ISSUE #5719: Should undefined theme elements throw a warning instead of an error?
This issue is a follow-up to #5702.
Instead of throwing errors, I think we should throw a warning and remove the offending element from the theme.

``` r
library(ggplot2)

ggplot(mpg, aes(displ, hwy)) +
  geom_point() +
  theme(foobar = "qux")
#> Error in `plot_theme()` at ggplot2/R/plot-build.R:178:3:
#> ! The `foobar` theme element is not defined in the element hierarchy.
```

<sup>Created on 2024-02-28 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by teunbrand**
This PR aims to fix #5719.

Briefly, when declaring an unknown theme element, we throw a warning instead of error. The offending theme element is then removed from the theme.

This is more gentle if you make a typo or for us if we want to deprecate theme arguments.

Reprex from the issue:

``` r
devtools::load_all("~/packages/ggplot2/")
#> ℹ Loading ggplot2

ggplot(mpg, aes(displ, hwy)) +
  geom_point() +
  theme(foobar = "qux")
#> Warning in plot_theme(plot): The `foobar` theme element is not defined in the
#> element hierarchy.
```

<sup>Created on 2024-03-05 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

File: tests/testthat/test-theme.R [hunk 1]
```diff
@@ -293,7 +293,7 @@ test_that("incorrect theme specifications throw meaningful errors", {
 test_that("element tree can be modified", {
   # we cannot add a new theme element without modifying the element tree
   p <- ggplot() + theme(blablabla = element_text(colour = "red"))
-  expect_snapshot_error(print(p))
```
Comment (thomasp85) [inline tests/testthat/test-theme.R:296]: Wouldn't this require an update of the snapshot

**Review by thomasp85 — APPROVED** [review]
Apart from the comment LGTM

File: tests/testthat/test-theme.R (hunk 1 above)
Comment (teunbrand) [reply tests/testthat/test-theme.R:296]: The message is exactly the same so the snapshot file hasn't changed

Comment (thomasp85) [reply tests/testthat/test-theme.R:296]: 👍

**Pull request merged**